js/main.js

```diff
@@ -606,20 +610,22 @@ class App extends React.Component {
     this.setState({layoutLists: layoutLists, layoutID: layoutID});
   }
 
-  broadcastKeyEvent = (event) => {
+  publishEvent = (event) => {
+    EventSystem.publish('global.event', event);
+  }
+
+  sendPaneMessage = (data) => {
```

Comment:
Comments added. As I was writing it, I realized that only focused panes should be calling this method (it adds `target` and `eid` of focused pane - not of the calling pane). I've done it this way to minimize changes to code (and I think for now it could be used), but maybe it's worth consideration not to add `target` and `eid` in this method, but just let the Pane fill it (it has paneId, but what with envId?)

Reply:
This is a really good point - I can see down the road it might be useful for panels that aren't focused to be sending events, however the current implementation is definitely workable for the events we have. A change to support those down the road will include telling panes what their env_id is, however that could all invoke a much harder expectation (like background panes from non-active environments being able to send/receive events, and we're definitely not there).